dapps/react-dapp-v2/src/components/Asset.tsx

```diff
@@ -56,16 +59,17 @@ interface AssetProps {
 
 const Asset = (props: AssetProps) => {
   const { asset } = props;
+  const decimals = asset.name === "KDA" ? 12 : undefined;
```

Comment:
Can we avoid the special handling for `KDA` here? Is there a reason for specifically 12 decimals in this case?

Reply:
When not specified the balance isn't correct because of what's returned from the `getKadenaBalanceForChain` function. I've altered the response from this function to be in line with the other chains, so the logic is contained within a "kadena-specific" file.